rados/ioctx.go

```diff
@@ -883,3 +883,77 @@ func (ioctx *IOContext) BreakLock(oid, name, client, cookie string) (int, error)
 		return int(ret), RadosError(int(ret))
 	}
 }
+
+// Snapshots
+// NOTE: self managed snapshots have not been implemented yet.
+
+// SnapHead represents the id of "no snapshot" used to resume normal operation when
+// reading and writing from a snapshot ioctx.
+const SnapHead uint64 = uint64(C.LIBRADOS_SNAP_HEAD)
```

Comment:
Specifying the type and a case is redundant. You can write this as: `const SnapHead = uint64(C.LIBRADOS_SNAP_HEAD)`.
Also, if these uint64 values are meant as opaque identifiers (eg you don't expect to do math on them). Perhaps it would be good to define a type such as `type SnapId uint64` or `type SnapIndex` something like that.

Reply:
Looking at librados.h this is pretty much what ceph does in C. Let's do the same as I suggested previously.